drivers/regulator/axp803.c

```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <error.h>
+#include <i2c.h>
+#include <mmio.h>
+#include <regulator/axp803.h>
+
+#define SYNCHRONOUS_CONTROL		0x14
+#define OUTPUT_POWER_CONTROL_1	0x10
+#define OUTPUT_POWER_CONTROL_2	0x12
+#define OUTPUT_POWER_CONTROL_3	0x13
+#define DLDO1_VOLTAGE_CONTROL	0x15
+#define DLDO2_VOLTAGE_CONTROL	0x16
+#define DLDO3_VOLTAGE_CONTROL	0x17
+#define DLDO4_VOLTAGE_CONTROL	0x18
+#define ELDO1_VOLTAGE_CONTROL	0x19
+#define ELDO2_VOLTAGE_CONTROL	0x1A
+#define ELDO3_VOLTAGE_CONTROL	0x1B
+#define FLDO1_VOLTAGE_CONTROL	0x1C
+#define FLDO2_VOLTAGE_CONTROL	0x1D
+#define DCDC1_VOLTAGE_CONTROL	0x20
+#define DCDC2_VOLTAGE_CONTROL	0x21
+#define DCDC3_VOLTAGE_CONTROL	0x22
+#define DCDC4_VOLTAGE_CONTROL	0x23
+#define DCDC5_VOLTAGE_CONTROL	0x24
+#define DCDC6_VOLTAGE_CONTROL	0x25
+#define ALDO1_VOLTAGE_CONTROL	0x28
+#define ALDO2_VOLTAGE_CONTROL	0x29
+#define ALDO3_VOLTAGE_CONTROL	0x2A
+
+static int
+axp803_disable(struct device *dev, char* channel)
+{
+	uint8_t reg;
+	int status;
+
+	if(0 == strcmp(channel, "DLDO1")){
```

Comment:
I would recommend creating an `enum` for all of these cases and also create a helper function to eliminate the redundant code once you have a proof of concept.